### PR TITLE
fix: fixes #76, adding more error checking and unifying calculation returns

### LINKED ISSUE #76: Nudge calculation error checking is erroneous since Eva v4
See: https://github.com/automata-tech/eva_python_sdk/blob/9674667bc2883d271731dcbbf273712f0ab4b178/evasdk/eva_http_client.py#L484 and https://docs.automata.tech/v/4.6.0/api/migration-guide/#upgrading-to-400

Starting with Eva v4, nudge won't always return a `nudge.joints` property on success, we need to make a change to the SDK to either handle the MotionError through exceptions or change the return of the function (breaking change). Consider doing the same for `rotate`, `inverse_kinematics`, etc.

### PR DESCRIPTION
**Pull request by LouisBrunner**
Remove the FK's `fk_type`, add more error checking following 4.0.0's new calculation error format, make nudge/rotate/IK consistent in their return format.

This is a breaking change.

### CODE REVIEW

File: evasdk/eva_http_client.py [hunk 1]
```diff
@@ -500,8 +526,7 @@ def calc_inverse_kinematics(self, guess, target_position, target_orientation, tc
 
         if r.status_code != 200:
             eva_error('inverse_kinematics error', r)
-        return r.json()
-
+        return self.__check_calculation(r, 'calc_inverse_kinematics', 'ik')['joints']
```
Comment (tallpress) [inline evasdk/eva_http_client.py:529]: for this `'calc_inverse_kinematics'` and similar `'calc_forward_kinematics'`, `'calc_nudge'` and `'calc_rotate'` could use `<function>.__name__`?

**Review by tallpress — APPROVED** [review]
LGTM, one nit/small suggestion

File: evasdk/eva_http_client.py (hunk 1 above)
Comment (LouisBrunner) [reply evasdk/eva_http_client.py:529]: You mean like this?
```
return self.__check_calculation(r, self.calc_inverse_kinematics.__name__, 'ik')['joints']
```

Comment (tallpress) [reply evasdk/eva_http_client.py:529]: I meant like that yea, but now you typed it out seems a bit much ado about nothing... lol. Maybe enum would be nicer 🤷‍♂️. But I don't think this is important

**Review by tallpress — DISMISSED** [review]

**Review by tallpress — APPROVED** [review]

**Pull request merged**